server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2224,6 +2225,10 @@ private IndexWriterConfig getIndexWriterConfig() {
                 new SoftDeletesRetentionMergePolicy(Lucene.SOFT_DELETES_FIELD, softDeletesPolicy::getRetentionQuery,
                     new PrunePostingsMergePolicy(mergePolicy, IdFieldMapper.NAME)));
         }
+        // We wrap the merge policy for all indices even though it is mostly useful for time-based indices
+        // but there should be no overhead for other type of indices so it's simpler than adding a setting
+        // to enable it.
+        mergePolicy = new ShuflleForcedMergePolicy(mergePolicy);
```

Comment:
I agree with doing it all the time for simplicity, but can you add an escape hatch in case it proves problematic for some use-cases?

Reply:
Sure I added a system property in https://github.com/elastic/elasticsearch/pull/48533/commits/aae5c3013388034f0fb9e3d6d9beaa50c0f6e09a